.travis.yml

```diff
@@ -22,13 +22,13 @@ before_script:
 
   # macOS dependencies
   - if [ "$TRAVIS_OS_NAME" = "osx" ]; then brew update; fi
-  - if [ "$TRAVIS_OS_NAME" = "osx" ]; then brew tap robotology/cask; fi
-  - if [ "$TRAVIS_OS_NAME" = "osx" ]; then brew install --only-dependencies yarp; fi
-  - if [ "$TRAVIS_OS_NAME" = "osx" ]; then brew install gsl sdl sdl_gfx sdl_image homebrew/science/ipopt; fi
   # Fix for opencv and numpy already installed by system
   - if [ "$TRAVIS_OS_NAME" = osx ]; then rm '/usr/local/bin/f2py'; fi
   - if [ "$TRAVIS_OS_NAME" = osx ]; then rm -r '/usr/local/lib/python2.7/site-packages/numpy'; fi
-  - if [ "$TRAVIS_OS_NAME" = "osx" ]; then brew install opencv; fi
+  # brew dependencies
+  - if [ "$TRAVIS_OS_NAME" = "osx" ]; then brew tap robotology/cask; fi
+  - if [ "$TRAVIS_OS_NAME" = "osx" ]; then brew install --only-dependencies yarp; fi
+  - if [ "$TRAVIS_OS_NAME" = "osx" ]; then brew install --only-dependencies icub-main --with-ipopt --with-opencv; fi
```

Comment:
One last thing, beware that the `opencv` formulae has been [greatly rewritten](https://github.com/Homebrew/homebrew-core/blob/master/Formula/opencv.rb) and now provides only a minimum set of options.

This has been made by the homebrew maintainers to reduce the clutter in their repo issues of people inquiring for help due to problem unrelated to the formula itself.

If we need particular options in OpenCV we must design our own formula, [like I did](https://github.com/claudiofantacci/homebrew-formulae) for my setup.

Reply:
Thanks to pointing that out.
I don't know which kind of features we require in `opencv`.  For sure, before, we were using the default formula.

Do you think it makes sense to add the custom opencv [here](https://github.com/robotology/homebrew-formulae/) ?

Reply:
Maybe not. My suggestion is to keep track/in mind this issue and to eventually take action.

The current `opencv` formula is good enough for a **personal setup**, it uses `WITH_EIGEN`, `WITH_FFMPEG`, `WITH_OPENEXR` and **with** `opencv_contrib`. Some other options are forced to `OFF`, like `WITH_OPENGL`, `WITH_QT` and `WITH_CUDA`. While the latter is generally required, the first two are instead of some help, but still, maybe not strictly required by `icub-main`.

The correct question is: _for a personal, basic, icub-main setup, what do a user need from OpenCV?_

cc @drdanz @pattacini @barbalberto @vtikha

Reply:
As long as compilation (and tests, if present) are working in Travis, I think this PR is good to go. The check on the opencv option seems to be an issue of the `icub-main` formula in https://github.com/robotology/homebrew-formulae , not a specific issue of this PR.

Reply:
Fair enough @traversaro. I just wanted to bring this into our attention. Just in case.

If something happens, we will divert our thoughts on `homebrew-formulae` repo.